lldb/source/Target/Target.cpp

```diff
@@ -3098,6 +3107,23 @@ void Target::SetAllStopHooksActiveState(bool active_state) {
   }
 }
 
+// FIXME:  Ideally we would like to return a `const &` (const reference) instead
```

Comment:
Are you sure that using the iterator of the map will preserve the order of the stop-hooks in this listing?  I'm not concerned about internal stop-hooks, but it wouldn't be good if you added a bunch of stop hooks, then deleted some, and then the order of the stop-hooks in the listing got scrambled.
Might be good to add a test to ensure that this ordering is maintained to ensure that.

Reply:
Added new test to pin down the behavior: 27b1441d3e27a12a4
Made sure it goes red when order is messed up, for example:
```
for (auto &[_, hook] : llvm::reverse(m_stop_hooks))
    stop_hooks.push_back(hook);
```